CMakeLists.txt

```diff
@@ -60,12 +64,23 @@ include_directories(osal)
 include_directories(osal/${OS})
 include_directories(oshw/${OS})
 
-add_library(soem STATIC
-  ${SOEM_SOURCES}
-  ${OSAL_SOURCES}
-  ${OSHW_SOURCES}
-  ${OSHW_EXTRA_SOURCES})
-target_link_libraries(soem ${OS_LIBS})
+
+
+if(BUILD_SHARED_LIBS)
```

Comment:
Is this needed? Seems like it would suffice to just set BUILD_SHARED_LIBS to TRUE.
https://cmake.org/cmake/help/v3.0/variable/BUILD_SHARED_LIBS.html#variable:BUILD_SHARED_LIBS

Reply:
You had `add_library(soem STATIC`

If the word STATIC is removed, then yes, it suffices to set that option to TRUE, it is global